packages/mysql/lib/mysql_p.js

```diff
@@ -37,7 +37,7 @@ function patchCreateConnection(mysql) {
 
   mysql['createConnection'] = function patchedCreateConnection() {
     var connection = mysql[baseFcn].apply(connection, arguments);
-    if (connection.then instanceof Function) {
+    if (connection != null && connection.then instanceof Function) {
```

Comment:
I think just a truthy check would suffice for all of these. With this code if `connection` is `undefined`, it will still throw an error.

Reply:
Truthy check will do as well, and it's probably easier to read, will refactor it. 👍 

However, since it was a loose equality check undefined wouldn't pass either as undefined and null are loosely equal. See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness (JavaScript works in mysterious ways 😛)

Reply:
Oh I see, I read that as a !== instead of !=.